clarify when it's ok to TLS only (#223)
ietf-rats-wg/draft-ietf-rats-msg-wrap: draft-ietf-rats-msg-wrap.md

## CMW Protection

CMW Records, Tags and Collections alone don't offer authenticity, integrity protection, or confidentiality.
It is the responsibility of the designer for each use case to determine the necessary security properties and implement them accordingly.
In some scenarios, for instance, if CMW encapsulates Reference Values or Endorsements, a secure channel (e.g., via TLS {{?BCP195}}) or object-level security (e.g., using JWT {{?BCP225}}) may suffice, but this is not always the case.

> When a CMW is used to encapsulate Evidence for composite or layered attestation of a single device, all components within the CMW must be cryptographically bound to prevent an attacker from replacing Evidence from a compromised device with that from a non-compromised device.

Reviewer: Is this a normative MUST? I don't think that's necessarily what we're going to see, even if it is unfortunate. The binding is up to the protocol, and not this envelope format.
```suggestion
When a CMW is used to encapsulate Evidence for composite or layered attestation of a single device, all components within the CMW must be cryptographically bound together to prevent an attacker from replacing Evidence from a compromised device with that from a non-compromised device.
```

CMW itself does not provide any mechanisms for authenticity, integrity protection, or confidentiality.
It is the responsibility of the designer for each use case to determine the necessary security properties and implement them accordingly.

> In some scenarios, for instance, if CMW encapsulates Reference Values and/or Endorsements, a secure channel (e.g., via TLS {{?BCP195}}) or object-level security (e.g., using JWT {{?BCP225}}) may suffice, but this is not always the case.

Reviewer 2: ```suggestion
In some scenarios, for instance, if CMW encapsulates Reference Values or Endorsements, a secure channel (e.g., via TLS {{?BCP195}}) or object-level security (e.g., using JWT {{?BCP225}}) may suffice, but this is not always the case.
```

Reviewer 2: CMW can be signed. Wouldn't signed CMW address integrity protection and authenticity without needing TLS?

Reviewer 2: Given TLS addresses confidentiality, replay, as well as integrity and authenticity. What other threat is remaining?

CMW itself does not provide any mechanisms for authenticity, integrity protection, or confidentiality.
It is the responsibility of the designer for each use case to determine the necessary security properties and implement them accordingly.

> In some scenarios, for instance, if CMW encapsulates Reference Values or Endorsements, a secure channel (e.g., via TLS {{?BCP195}}) or object-level security (e.g., using JWT {{?BCP225}}) may suffice, but this is not always the case.

Reviewer 3: ```suggestion
Mechanisms to protect CMW, such as encapsulates Reference Values or Endorsements include secure channels (e.g., via TLS {{?BCP195}}) or object-level security (e.g., using JWT {{?BCP225}}).
```

Author: Thanks for the suggestion, but I don’t think it increases clarity.

> When a CMW is used to carry Evidence for composite or layered attestation of a single device, all components within the CMW must be cryptographically bound to prevent an attacker from replacing Evidence from a compromised device with that from a non-compromised device.

Reviewer 3: ```suggestion
When a CMW is used to encapsulate Evidence or Attestation Results, e.g., for composite or layered attestation of a single device, all components within the CMW must also be cryptographically bound to prevent an attacker from replacing Evidence or Attestation Results from a compromised device with that from a non-compromised device.
```

> CMW itself does not provide any mechanisms for authenticity, integrity protection, or confidentiality.

Reviewer 2: Wouldn't the signed-cmw satisfy the authenticity and integrity protection properties?

Author: Added in [d26fe34](https://github.com/ietf-rats-wg/draft-ietf-rats-msg-wrap/pull/223/commits/d26fe34b6ad1f7c355dd02280e8bd79b8ca6d21c)
